File: frontend/lib/schemas/provenance.ts
// Zod schema for GET /api/v1/provenance/{dataset}/{model} — mirrors backend ProvenanceResponse.

import { z } from "zod";

export const ProvenanceResponseSchema = z.object({
  dataset_slug: z.string(),
  model_name: z.string(),
  model_version: z.string(),
  git_sha: z.string(),
  created_at: z.string().refine((s) => Number.isFinite(Date.parse(s)), {
    message: "expected ISO-8601 datetime string",
  }),
  output_parquet_key: z.string(),
});

export type ProvenanceResponse = z.infer<typeof ProvenanceResponseSchema>;

File: backend/app/schemas/provenance.py
# Response schema for GET /api/v1/provenance/{dataset_slug}/{model_name} (latest precompute run).

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProvenanceResponse(BaseModel):
    """Latest precompute run metadata for a dataset × model pair."""

    model_config = ConfigDict(from_attributes=True)

    dataset_slug: str
    model_name: str
    model_version: str
    git_sha: str
    created_at: datetime
    output_parquet_key: str
